packages/emitter-framework/test/typescript/components/union-declaration.test.tsx

```diff
@@ -1,182 +1,379 @@
-        const union = Array.from((namespace as Namespace).unions.values())[0];
+        const { TestUnion } = (await runner.compile(`
+          namespace DemoService;
+          @test union TestUnion {
+            one: "one",
+            two: "two"
+          }
+        `)) as { TestUnion: Union };
 
         const res = render(
-          <Output program={program}>
+          <Output program={runner.program}>
             <SourceFile path="test.ts">
-              <UnionDeclaration type={union}>"three"</UnionDeclaration>
+              <UnionDeclaration type={TestUnion}>"three"</UnionDeclaration>
             </SourceFile>
           </Output>,
         );
 
-        const testFile = res.contents.find((file) => file.path === "test.ts");
-        assert(testFile, "test.ts file not rendered");
-        const actualContent = await format(testFile.contents as string, { parser: "typescript" });
-        const expectedContent = await format(`type TestUnion = "one" | "two" | "three"`, {
-          parser: "typescript",
-        });
-        expect(actualContent).toBe(expectedContent);
+        assertFileContents(
+          res,
+          d`
+            type TestUnion = "one" | "two" | "three";
+          `,
+        );
       });
 
-      it("renders an union expression", async () => {
-        const program = await getProgram(`
+      it("renders a union expression", async () => {
+        const { TestUnion } = (await runner.compile(`
           namespace DemoService;
-          union TestUnion {
+          @test union TestUnion {
             one: "one",
             two: "two"
           }
-          `);
+        `)) as { TestUnion: Union };
+
+        const res = render(
+          <Output program={runner.program}>
+            <SourceFile path="test.ts">
+              let x: <UnionExpression type={TestUnion} /> = "one";
+            </SourceFile>
+          </Output>,
+        );
+
+        assertFileContents(
+          res,
+          d`
+            let x: "one" | "two" = "one";
+          `,
+        );
+      });
 
-        const [namespace] = program.resolveTypeReference("DemoService");
-        const union = Array.from((namespace as Namespace).unions.values())[0];
+      describe("Discriminated Union", () => {
+        it("renders a discriminated union", async () => {
+          const { TestUnion } = (await runner.compile(`
+            namespace DemoService;
+            @discriminated
+            @test union TestUnion {
+              one: { oneItem: true },
+              two: true
+            }
+          `)) as { TestUnion: Union };
+
+          const res = render(
+            <Output program={runner.program}>
+              <SourceFile path="test.ts">
+                <UnionDeclaration type={TestUnion} />
+              </SourceFile>
+            </Output>,
+          );
+
+          assertFileContents(
+            res,
+            d`
+              type TestUnion = {
```

Comment:
This formatting is trash but that's not your problem. I'll fix this in alloy at some point ☺️